### DISCUSSION TITLE
Discussion for the biweekly meeting (2022/08/31)

### DISCUSSION THREAD

**Discussion by sczyh30** [opensergo/opensergo-specification#36] [General]
Discussion for the biweekly meeting of the OpenSergo community (2022/08/31)

---

本次 OpenSergo 双周会将于 2022/08/31（周三）进行，欢迎大家预留时间参与，也欢迎大家发起 topic 来分享。议程如下：

* 同步上一周期的进展，包括 spec 建设、Control Plane && SDK 进展、社区生态接入等进展
* 社区分享与讨论
  * OpenSergo Control Plane 及 SDK 整体方案介绍
  * OpenSergo + Sentinel Java 演示
  * 流量路由、数据库治理 spec 细节对焦
  * 其它社区分享讨论

主题：OpenSergo 双周会（钉钉群会议）
时间：2022年8月31日 周三 19:30 - 21:00
入会口令：244 801 51010
电话呼入：+862281944261（中国大陆）+867388953916（中国大陆）
入会链接：dingtalk://dingtalkclient/page/videoConfFromCalendar?confId=65d3c04b-b69e-44ce-b310-35054cff1d57&calendarId=1881630426

上次社区双周会记录：https://github.com/opensergo/opensergo-specification/discussions/27

**Comment by sczyh30** [comment]
* [WIP] Initial version of OpenSergo Java SDK: https://github.com/opensergo/opensergo-java-sdk/pull/1
* [WIP] Sentinel OpenSergo data-source (Java): https://github.com/alibaba/Sentinel/issues/2841
* [WIP] Proto of OpenSergo universal transport service: https://github.com/opensergo/opensergo-proto/pull/2

**Comment by xzd1621** [comment]
在 OpenSergo Control Plane 内部，主要包括CRD watcher、CRD transformer、gRPC server 这几个部分。CRD watcher主要用于监听CRD变化；gPRC server主要用于接收订阅请求和下发配置，以及接入实例的连接管理；CRD transformer 主要用于对配置做一些校验和转换处理，将面向用户的 CRD 转换为面向传输层的 proto。

<img width="1093" alt="image" src="https://user-images.githubusercontent.com/30654585/187685459-eee43b2c-92ea-450f-8566-1571d0470e68.png">

**Comment by sczyh30** [comment]
本周 OpenSergo 社区双周会纪要：

1. 社区同学 @xzd1621  分享了 OpenSergo Control Plane、SDK 的整体设计，并分享了控制面与数据面之间 gRPC 协议的详细设计。Control Plane 的初版代码还在整理中，整理完成后将正式开源透出。

* OpenSergo universal transport service gRPC 协议设计：https://github.com/opensergo/opensergo-proto/issues/1
* Control Plane 设计：https://github.com/opensergo/opensergo-specification/discussions/36#discussioncomment-3517830

2. 社区通过 Sentinel (Java) + OpenSergo 的示例，演示了从拉起控制面，到应用通过 Sentinel 数据源接入 OpenSergo 控制面，再到通过控制面配置流控降级与容错 CRD 使得接入应用可以生效限流 整个流程。目前 OpenSergo Java SDK 及 Sentinel 数据源均已完成初版，可以作为社区接入的参考，欢迎社区一起来 review。

* Initial version of OpenSergo Java SDK: https://github.com/opensergo/opensergo-java-sdk/pull/1
* Sentinel OpenSergo data-source (Java): https://github.com/alibaba/Sentinel/pull/2842

3. 社区进行了一些细节的讨论，包括 spec 的扩展性、控制面与数据面之间通信协议的扩展性，以及未来可以落地的生态。社区欢迎各个开发者、社区与企业一起来完善 OpenSergo spec，共同扩充微服务治理标准与场景；同时对于 SDK 及 Control Plane 的扩展性，欢迎社区一起参与设计和代码贡献。

**Comment by sczyh30** [comment]
关于数据库治理 spec，这两天也会与 ShardingSphere 社区同学 @mlycore 进行进一步对焦，预计下周整理完成初版。
